lib/collection/src/shards/replica_set/shard_transfer.rs

```diff
@@ -207,4 +233,61 @@ impl ShardReplicaSet {
             )))
         }
     }
+
+    /// Send all queue proxy updates to remote and transform into forward proxy
+    ///
+    /// When a queue or forward proxy shard needs to be unproxified into a local shard again we
+    /// typically don't have room to handle errors. A queue proxy shard may error if it fails to
+    /// send updates to the remote shard, while a forward proxy does not fail at all when
+    /// transforming.
+    ///
+    /// This method allows to transfer queued updates before the shard is unproxified. This allows
+    /// for proper error handling at the time this method is called. Because the shard is
+    /// transformed into a forward proxy after this operation it will not error again when the
+    /// shard is eventually unproxified again.
+    ///
+    /// If the local shard is a queue proxy:
+    /// - Transfers all missed updates to remote
+    /// - Transforms queue proxy into forward proxy
+    ///
+    /// Does nothing if the local shard is not a queue proxy.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if transferring all updates to the remote failed.
+    pub async fn queue_proxy_into_forward_proxy(&self) -> CollectionResult<()> {
+        // First pass: transfer all missed updates with shared read lock
+        {
+            let local_read = self.local.read().await;
+            if let Some(Shard::QueueProxy(proxy)) = &*local_read {
+                proxy.transfer_all_missed_updates().await?;
+            } else {
+                return Ok(());
+            }
```

Comment:
Sorry, but I just love let-else statements. Feels cleaner, IMO.

```suggestion
            let Some(Shard::QueueProxy(proxy)) = &*local_read else {
                return Ok(());
            };
            
            proxy.transfer_all_missed_updates().await?;
```